src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -65,7 +68,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     }
 
     String query = encodeValue(param);
-    List<String> allLinks = new ArrayList<>();
+    List<Card> allCards = new ArrayList<>();
 
     // TODO: after implementing scraping, consider changing getLink calls to a
     // for-loop
```

Comment:
Food for thought for when you consolidate the calls into a for-loop, I think it would make sense to create an interface for your Clients, something like:
```
interface Client {
  public String getCseId();
  public Card search();
}
```
And then have each of your W3School, StackOverflow, and GeeksForGeeks clients implement your `Client` interface. That way, you have a standard API for each of your clients and you can just iterate through a list of clients:
```
List<Client> clients = new ArrayList<Client>();
clients.add(new W3SchoolClient());
...

for (Client client : clients)
{
  String link = getLink(client.getCseId(), query);
  Card card = client.search(link);
  ...
}
```
Obviously, don't need to do this in this PR but something to keep in mind. I'll add a Github task for this.